src/web/components/ClickToView.tsx

```diff
@@ -89,74 +113,115 @@ const Headline = ({
 
 const Body = ({
 	children,
-	width,
+	role,
 }: {
 	children: React.ReactNode;
-	width: number;
-}) => (
-	<div
-		className={css`
-			${width > 300 ? textSans.small() : textSans.xsmall()}
-			a {
-				${width > 300 ? textSans.small() : textSans.xsmall()}
-			}
-		`}
-	>
-		{children}
-	</div>
-);
+	role: RoleType;
+}) => {
+	const textSize = roleTextSize(role);
+	return (
+		<div
+			className={css`
+				${textSize}
+				a {
+					${textSize}
+				}
+			`}
+		>
+			{children}
+		</div>
+	);
+};
+
+const AcceptButton = ({
+	role,
+	handleClick,
+}: {
+	role: RoleType;
+	handleClick: Function;
+}) => {
+	return (
+		<div>
+			<Button
+				priority="primary"
+				size={roleButtonSize(role)}
+				icon={<SvgCheckmark />}
+				iconSide="left"
+				onClick={() => handleClick()}
+			>
+				{roleButtonText(role)}
+			</Button>
+		</div>
+	);
+};
 
-export const ClickToView = ({ children, width, height, onAccept }: Props) => {
+export const ClickToView = ({
+	children,
+	role,
+	onAccept,
+	isTracking,
+	source,
+	sourceDomain,
+}: Props) => {
 	const [showOverlay, setShowOverlay] = useState<boolean>(true);
 
 	const handleClick = () => {
 		setShowOverlay(false);
-		onAccept();
+		if (onAccept) {
+			setTimeout(() => onAccept());
+		}
 	};
 
-	if (showOverlay) {
+	let headlineText;
+	let body;
+
+	const roleWithDefault: RoleType = role || 'inline';
```

Comment:
We can replace this with `role = 'inline'` in the props on line 160. But maybe we can remove it completely by making `role` required? It _is_ required and should always be provided. If the parent component really doesn't know then that might be the better place to default to 'inline'

Reply:
Souonds good.

Worth noting that the concept of `role` changed recently. There was a change in the CAPI code to add this property to a lot more elements by default and a reindex of all the articles was run to bring it though. But even then, perhaps it's not that common?